pkg/server/biz/hook/hook.go

```diff
@@ -0,0 +1,188 @@
+package hook
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+
+	"github.com/caicloud/cyclone/pkg/apis/cyclone/v1alpha1"
+	"github.com/caicloud/cyclone/pkg/meta"
+	api "github.com/caicloud/cyclone/pkg/server/apis/v1alpha1"
+	in "github.com/caicloud/cyclone/pkg/server/biz/integration"
+	"github.com/caicloud/cyclone/pkg/server/biz/scm"
+	"github.com/caicloud/cyclone/pkg/server/common"
+	"github.com/caicloud/cyclone/pkg/server/config"
+	"github.com/caicloud/cyclone/pkg/server/handler"
+	"github.com/caicloud/cyclone/pkg/util/cerr"
+	"github.com/caicloud/nirvana/log"
+)
+
+var o *Operator
+
+// Operator
+type Operator struct {
+	mutex sync.Mutex
+}
+
+// GetInstance returns the Operator instance
+func GetInstance() *Operator {
+	if o == nil {
+		o = &Operator{
+			mutex: sync.Mutex{},
+		}
+	}
+
+	return o
+}
+
+// ListWfts list all relative workflow triggers
+func (*Operator) ListWfts(tenant, repo, integration string) (*v1alpha1.WorkflowTriggerList, error) {
+	labelMap := make(map[string]string)
+	if repo != "" {
+		labelMap[meta.LabelRepoOfWft] = encodeRepoValue(repo)
+	}
+	if integration != "" {
+		labelMap[meta.LabelIntegrationOfWft] = integration
+	}
+
+	listOption := metav1.ListOptions{
+		LabelSelector: labels.Set(labelMap).String(),
+	}
+
+	wfts, err := handler.K8sClient.CycloneV1alpha1().WorkflowTriggers(common.TenantNamespace(tenant)).List(listOption)
+	if err != nil {
+		return nil, err
+	}
+
+	return wfts, nil
+}
+
+// RegisterSCMWebhook registers SCM webhook if if has not been registered.
+func (*Operator) RegisterSCMWebhook(tenant, wftName, secretName, repo string) error {
```

Comment:
I'd prefer to describe a interface like `Manager` (under package `hook`). With methods like `Register(...)`, `UnRegister(...)`.

Then make implementation like `SCMManager`...